PWGHF/D2H/Tasks/taskCharmPolarisation.cxx

```diff
@@ -217,20 +326,210 @@ struct TaskPolarisationCharmHadrons {
 
   }; // end init
 
+  /// \param invMassCharmHad is the invariant-mass of the candidate
+  /// \param ptCharmHad is the pt of the candidate
+  /// \param pzCharmHad is the pz of the candidate
+  /// \param rapCharmHad is the rapidity of the candidate
+  /// \param cosThetaStar is the cosThetaStar of the candidate
+  /// \param outputMl is the array with ML output scores
+  /// \param isRotatedCandidate is a flag that keeps the info of the rotation of the candidate for bkg studies
+  /// \param origin is the MC origin
+  template <charm_polarisation::DecayChannel channel, bool withMl, bool doMc, charm_polarisation::CosThetaStarType cosThetaStarType>
+  void fillRecoHistos(float& invMassCharmHad, float& ptCharmHad, float& pzCharmHad, float& rapCharmHad, float& cosThetaStar, std::array<float, 3>& outputMl, int& isRotatedCandidate, int8_t origin)
```

Comment:
```suggestion
  void fillRecoHistos(float invMassCharmHad, float ptCharmHad, float pzCharmHad, float rapCharmHad, float cosThetaStar, std::array<float, 3> outputMl, int isRotatedCandidate, int8_t origin)
```

similar comment done by Vit in this PR https://github.com/AliceO2Group/O2Physics/pull/5057 ("Passing small types by value is better")